benches/languages.rs

```diff
@@ -207,6 +208,62 @@ fn benchmark_parse_syslog(c: &mut Criterion) {
     benchmark_configs(c, "parse_syslog", configs, "in", "last", input, output);
 }
 
+fn benchmark_multifaceted(c: &mut Criterion) {
+    let configs: Vec<(&str, &str)> = vec![
+        (
+            "remap",
+            indoc! {r#"
+                [transforms.last]
+                  type = "remap"
+                  inputs = ["in"]
+                  source = """
+                  . = parse_syslog!(.message)
+                  .timestamp = format_timestamp!(.timestamp, format: "%c")
+                  """
+            "#},
+        ),
+        (
+            "lua",
+            indoc! {r#"
+                [transforms.last]
+                  type = "lua"
+                  inputs = ["in"]
+                  version = "2"
+                  source = """
+                  local function parse_syslog(message)
+                    local pattern = "^<(%d+)>(%d+) (%S+) (%S+) (%S+) (%S+) (%S+) (%S+) (.+)$"
+                    local priority, version, timestamp, host, appname, procid, msgid, sdata, message = string.match(message, pattern)
+
+                    return {priority = priority, version = version, timestamp = timestamp, host = host, appname = appname, procid = procid, msgid = msgid, sdata = sdata, message = message}
+                  end
+
+                  function process(event, emit)
+                    event.log = parse_syslog(event.log.message)
+                    emit(event)
+                  end
+                  """
+                  hooks.process = "process"
+            "#},
+        ),
+        (
+            "wasm",
+            indoc! {r#"
+                [transforms.last]
+                  type = "wasm"
+                  inputs = ["in"]
+                  module = "tests/data/wasm/multifaceted/target/wasm32-wasi/release/multifaceted.wasm"
+                  artifact_cache = "target/artifacts/"
+            "#},
+        ),
+    ];
+
+    let input = r#"<12>3 2020-12-19T21:48:09.004Z initech.io su 4015 ID81 - TPS report missing cover sheet"#;
```

Comment:
Would it be worth having more than one line of input for this, or randomized data?

Reply:
I'm not sure that randomness would achieve much here. The benchmarks do check for correctness (note the `assert_eq!(output, actual)` in the bench function) and I think we'd basically need to re-write the benchmarking logic to allow for randomization. That's a potentially worthy project but I think out of scope here.

Reply:
The randomness would be to make it more like "real" data, where the CPU isn't taking *exactly* the same branches each time, making it seem faster than it would be IRL and biasing the comparison towards whichever code predicts best. One step better without randomness would be to have an array of lines to loop through.

Reply:
I think having more than one line of input would be desirable, but there is a tension between using random data and consistency in benchmarks (also in asserting correct transformations). The benchmark scaffolding will need to be updated here to permit multiple lines though.

Reply:
I would like to make this more robust at some point. For now, though, the goal here is mostly to provide a realistic-ish comparison for an upcoming blog post. I think that these benchmarks do accomplish the modest goal of comparing across VRL, Wasm, and Lua for a few simple use cases.